admin/app/pagepresser/HtmlCleaner.scala

```diff
@@ -120,56 +52,76 @@ object BasicHtmlCleaner extends HtmlCleaner {
       }
     }.mkString("&")
   }
-
-  def removeAds(document: Document): Document = {
-    val element = document.getElementById("sub-header")
-
-    if (element != null) {
-      val ads = element.children().toList.filterNot(e => e.attr("class") == "top-navigation twelve-col top-navigation-js")
-      ads.foreach(_.remove())
-
-      val htmlComments = element.childNodes().filter(node => node.nodeName().equals("#comment"))
-      htmlComments.foreach(_.remove())
-
-      val promo = document.getElementById("promo")
-      if(promo != null) promo.remove()
-    }
-
-    document
-  }
-
   def removeRelatedComponent(document: Document): Document = {
     val element = document.getElementById("related")
     if(element != null) element.remove()
     document
   }
-
-  private def removeByClass(document: Document, className: String): Document = {
+  def removeByClass(document: Document, className: String): Document = {
     document.getElementsByClass(className).foreach(_.remove())
     document
   }
-
-  private def removeByTagName(document: Document, tagName: String): Document = {
+  def removeByTagName(document: Document, tagName: String): Document = {
     document.getElementsByTag(tagName).foreach(_.remove())
     document
   }
+}
 
-  private def addSwfObjectScript(document: Document): Document = {
+object BasicHtmlCleaner extends HtmlCleaner {
 
-    val swfScriptOpt = try {
-      val source = Source.fromInputStream(getClass.getClassLoader.getResourceAsStream("resources/r2/interactiveSwfScript.js"), "UTF-8").getLines().mkString
-      Some(source)
+  override def canClean(document: Document) = {
+    document.getElementsByAttribute("data-poll-url").isEmpty
+  }
 
-    } catch {
-      case ex: Exception => {
-        log.error(ex.getMessage)
-        None
-      }
+  override def clean(document: Document) = {
+    if (canClean(document)) {
+      basicClean(document)
+      removeScripts(document)
+      createSimplePageTracking(document)
+      removeByTagName(document, "noscript")
+    } else {
+      document
+    }
+  }
+
+  override def extractOmnitureParams(document: Document): Map[String, Seq[String]] = {
+    val omnitureNoScript = document.getElementById("omnitureNoScript")
+    if (omnitureNoScript != null) {
+      parse(omnitureNoScript.getElementsByTag("img").attr("src")).query.paramMap
+    } else {
+      Map.empty
     }
-    swfScriptOpt.foreach { script =>
-      val html = "<script type=\"text/javascript\">" + script + "</script>"
-      document.head().append(html)
+  }
+
+  def basicClean(document: Document): Document = {
```

Comment:
Should this go in the abstract class? All the implementations call it and don't override it?